src/cli.js

```diff
@@ -611,15 +611,21 @@ var exports = {
     }
 
     var filename;
+    var lintStdinFile;
 
     // There is an if(filename) check in the lint() function called below.
     // passing a filename of undefined is the same as calling the function
     // without a filename.  If there is no opts.filename, filename remains
     // undefined and lint() is effectively called with 4 parameters.
     if (opts.filename) {
-      filename = path.resolve(opts.filename);
+      filename = path.relative(path.resolve('.'), path.resolve(opts.filename));
+      var ignores = !opts.ignores ? loadIgnores({ cwd: opts.cwd }) :
+                                  opts.ignores.map(function(target) {
+                                    return path.resolve(target);
+                                  });
+      lintStdinFile = (opts.useStdin && !isIgnored(filename, ignores));
     }
-    if (opts.useStdin && opts.ignores.indexOf(filename) === -1) {
+    if (lintStdinFile) {
```

Comment:
As written, this branch will only be taken in the presence of the `--filename`
argument. The CLI currently lints standard input even when that argument is not
specified, so we should preserve that behavior. It's surprising that the patch
passed our test suite in the current state. [I've pushed a regression test to
`master`](https://github.com/jshint/jshint/commit/71f2f1f08d007a9f1de0ceeaab4ebe08ef70b6f3),
so if you rebase your branch, you'll have an easier time verifying this.

Reply:
Rebased and fixed the bug